### ISSUE TITLE
ContractEception

### ISSUE THREAD

**Issue by ordinaryorange** [mfidemraizer/trackerdog#4]
Basic POCO class with List<T> property where T is also basic POCO class
Added parent POCO type to TrackerConfiguration as per docs.
Then call On AsTrackable() extension method and get ContractException (as below)
Occurs irrespective of whether property in question is virtual or not.
Using Ver 1.1.3 from NuGet

Exception message; 
Assertion failed: parentObjectProperty.PropertyType.GetGenericTypeDefinition().IsAssignableFrom(collectionImplementationDetail.Key)  Trackable collection implementation of type 'System.Collections.Generic.IList`1, mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089' cannot be set to the target property 'Castle.Proxies.MyClass.MyList' with type 'System.Collections.Generic.List`1[[UserQuery+myObject, query_fnvjno, Version=0.0.0.0, Culture=neutral, PublicKeyToken=null]], mscorlib, Version=4.0.0.0, Culture=neutral, PublicKeyToken=b77a5c561934e089'. This isn't supported because it might require a downcast. Please provide a collection change tracking configuration to work with the more concrete interface.

**Comment by mfidemraizer** [comment]
I see, confirm is this is the scenario:

```
public class A
{
      public List<A> ListOfA { get; set; }
}

A a = new A().AsTrackable();
```

Am I in the right track? If so, I'll implement a test case and I'll try to fix the whole issue!

Thank you in advance for notifying this issue!

**Comment by mfidemraizer** [comment]
I could reproduce the issue already. I'm already working on fixing it!

**Comment by ordinaryorange** [comment]
Bingo spot on.
Event if I change ListOfA to a List of B

public class B
{
	public int MyProperty { get; set; }
}
TrackerDogConfiguration.TrackTheseTypes(Track.ThisType<A>(), Track.ThisType<B>());
	var p = new A().AsTrackable();

Same error.

**Comment by mfidemraizer** [comment]
@ordinaryorange Ok. Thanks. I've already fixed the contract exception improving the contract itself. Now I'm on the other issue.

**Comment by ordinaryorange** [comment]
Cool. Glad to see it was an easy one.

**Comment by mfidemraizer** [comment]
There's still an issue with this. Once I've fixed the contract, now arises another problem. 

I found that only on current use case Castle DynamicProxy isn't generating a collection proxy which implements the interface of  the target collection.

That is, if we proxy a `List<F>` where `F` is the enclosing type, it generated a `IList`1Proxy` which doesn't implement `IList<T>`! I'm trying to fix this.
